### DISCUSSION TITLE
How to use it outside built-in simple editor to have intellisence in VS Code?

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [spakin/SimpInkScr#103] [Q&A] [answered]
What should I import in .py file to have API hints? I can't write scripts without having intellisence.

**Comment by spakin** [comment]
I have to admit, I've not tried writing Simple Inkscape Scripting scripts myself within VS Code.  Let me ask the user who contributed VS Code support (in #54 and #55):

@inter1965: Could you please field @EmilyGraceSeville7cf's question?  (And I hope I haven't broken VS Code support since you added it.)  Thanks.

**Comment by EmilyGraceSeville7cf** [comment, marked as answer]
@spakin, issue is solved via [this](https://stackoverflow.com/a/15268990) SO answer: I have to run this command in terminal: `pip3 install git+https://github.com/spakin/SimpInkScr#egg=simpinkscr` and then after installation do `import simpinkscr` to have intellisence enabled. Can you mention it in Wiki? Not everybody Python programmer and not everybody knows this trick.

**Reply by spakin** [reply]
Thanks for the information; I'll be glad to include this in the wiki.  I have two questions, though:
1. Instead of installing a separate version of Simple Inkscape Scripting with `pip`, would it suffice to set your `PYTHONPATH` environment variable to point to your existing Simple Inkscape Scripting directory?  For example, on my Linux system I can run
```bash
env PYTHONPATH=$HOME/.config/inkscape/extensions/simple_inkscape_scripting:/usr/share/inkscape/extensions:$PYTHONPATH python -c 'import simpinkscr'
```
and not receive any error messages.
2. When you write, "after installation do `import simpinkscr`", does that mean you have to include that line in every script you write?

**Reply by EmilyGraceSeville7cf** [reply]
> Instead of installing a separate version of Simple Inkscape Scripting with pip, would it suffice to set your PYTHONPATH environment variable to point to your existing Simple Inkscape Scripting directory?

I am not a Python developer, so I can't answer this question. :) But I guess it worth it to mention both ways to get intellisence in an external editor like VS Code. Mine (not exactly mine, but you get it 😄) and yours.

> When you write, "after installation do import simpinkscr", does that mean you have to include that line in every script you write?

It does. AFAIU, this import is implicitly included while writing in built-in editor.

**Reply by spakin** [reply]
If you use `import simpinkscr`, doesn't this mean you have to use `simpinkscr.circle`, `simpinkscr.rect`, etc. for all Simple Inkscape Scripting operations?  Wouldn't it be simpler—and more compatible with running Simple Inkscape Scripting from the Inkscape GUI—to use `from simpinkscr import *`?

**Reply by EmilyGraceSeville7cf** [reply]
> If you use `import simpinkscr`, doesn't this mean you have to use `simpinkscr.circle`, `simpinkscr.rect`, etc. for all Simple Inkscape Scripting operations?  Wouldn't it be simpler—and more compatible with running Simple Inkscape Scripting from the Inkscape GUI—to use `from simpinkscr import *`?

It would. :) I just use it for code clarity.

**Reply by spakin** [reply]
All right; so how do you like this description?

https://github.com/spakin/SimpInkScr/wiki/Running#visual-studio-code

**Reply by EmilyGraceSeville7cf** [reply]
> All right; so how do you like this description?
> 
> https://github.com/spakin/SimpInkScr/wiki/Running#visual-studio-code

I am completely fine with it.

**Reply by spakin** [reply]
Great!  Thanks for taking a look—and for reporting how to get Simple Inkscape Scripting to work with Visual Studio Code + IntelliSense.
